### PR TITLE
[TST] Tests and better debugging

### PR DESCRIPTION
**Pull request by rmarkello**
I'm working on adding some unit tests for `tedana.utils`! Unfortunately a large portion of the functions require data files to operate on (or, at least, would be much easier to test with data files).

Would it be reasonable to downsample some functional data from [OpenNeuro](https://openneuro.org/datasets/ds000210/versions/00002) and ship them with `tedana` in a `tests/data` directory? It'd be nice to be able to run some tests locally, especially considering `tedana` doesn't have any external software dependencies anymore!

### CODE REVIEW

**Comment by emdupre** [comment]
Certainly with io and utils, I think we could use down sampled data ! I worry that T2* map and similar functions might not be as resilient with a heavily down sampled data set, but it's something we could test !

**Comment by rmarkello** [comment]
This PR is working to address both #15 and #36. 

I think #42 should be merged first and I'll deal with the conflicts that will arise from renaming / working on the same stuff. I'll hold off on working on this too much more until that's settled.

**Comment by rmarkello** [comment]
Alright, I think this might be ready to go. Travis is now responsible for all testing except the primary integration test (`test_tedana.py`), which Circle handles. I added a few tests for `tedana.utils.utils`, but there's still a lot more to be covered. Nonetheless, at this point I think it might be best to integrate this into master and build from there in more targeted PRs.

This closes #36 and closes #45. It's a partial response to #15, but certainly not enough to warrant closing that issue. This may also close #13, with all the work that @tsalo has been doing merged in.

Let me know if you'd like to see anything else added to this, or if you have any comments!

File: tedana/info.py [hunk 1]
```diff
@@ -30,17 +30,21 @@
...
+    'pytest-cov'
 ]
 
 EXTRA_REQUIRES = {
-    'doc': ['sphinx>=1.5.3', 'sphinx_rtd_theme', 'sphinx-argparse'],
+    'doc': [
+        'sphinx>=1.5.3',
+        'sphinx_rtd_theme',
+        'sphinx-argparse'
+    ],
```
Comment (emdupre) [inline tedana/info.py]: Could we add `numpydoc` here ? Comes from #43

Comment (emdupre) [reply tedana/info.py]: Self-assigning this !

**Comment by emdupre** [comment]
I was thinking about waiting until we got closer to 80% at our 0.1.0
milestone. Do you think it's valuable to have beforehand ?

**Comment by tsalo** [comment]
It will be a little depressing at first, but at least for me it reminds me that there's something to aim for. Totally up to you though.

**Comment by emdupre** [comment]
Thanks so much @kirstieJane !!

Yes, I think we could add detail in #15 which might be nice to point new
contributors in the right direction. And we can open more re-factoring
issues, though it seems like having tests in place help for re-factoring,
so I'm not sure what's the right order to promote those !

File: README.md [hunk 2]
```diff
@@ -38,21 +43,20 @@ numpy
...
 
-Want to learn more about our plans for developing tedana ? Check out [our roadmap](https://github.com/ME-ICA/tedana/projects). Have a question, comment, or suggestion ? Open or comment on one of [our issues](https://github.com/ME-ICA/tedana/issues) !
+Want to learn more about our plans for developing `tedana` ?
+Check out [our roadmap](https://github.com/ME-ICA/tedana/projects).
+Have a question, comment, or suggestion ?
+Open or comment on one of [our issues](https://github.com/ME-ICA/tedana/issues) !
 
-We ask that all contributions to tedana respect our [code of conduct](https://github.com/ME-ICA/tedana/blob/master/Code_of_Conduct.md).
-
-### :earth_americas: Mozilla Global Sprint (10-11 May, 2018) :earth_africa:
```
Comment (emdupre) [inline README.md]: #mozsprint is over ! so sad 😢

File: tedana/selection/select_comps.py [hunk 3]
```diff
@@ -257,31 +264,40 @@ def selcomps(seldict, mmix, mask, ref_img, manacc, n_echos, t2s, s0, olevel=2,
...
-        db = DBSCAN(eps=.005 + ii * .005, min_samples=3).fit(fz.T)
+        eps = .005 + ii * .005
+        db = DBSCAN(eps=eps, min_samples=3).fit(fz.T)
 
         # it would be great to have descriptive names, here
+        # DBSCAN found at least three non-noisy clusters
         cond1 = db.labels_.max() > 1
+        # DBSCAN didn't detect more classes than the total # of components / 6
         cond2 = db.labels_.max() < len(nc) / 6
+        # not totally sure if 0 is a special label for DBSCAN
```
Comment (emdupre) [inline tedana/selection/select_comps.py]: self-assign: add a #TODO here

**Pull request merged**